src/lib/frontend/models.mli

```diff
@@ -38,15 +38,11 @@ type objective_value =
 
 type t = {
   propositional : Expr.Set.t;
-  constants : ModelMap.t;
-  functions : ModelMap.t;
-  arrays : ModelMap.t;
+  model : ModelMap.t;
   objectives: (Expr.t * objective_value) Util.MI.t;
-  terms_values : (Shostak.Combine.r * string) Expr.Map.t
+  terms_values : (Shostak.Combine.r * Hstring.t) Expr.Map.t
```

Comment:
Not sure why we need the values to be interned `Hstring.t`? Also, shouldn't we be using `Value.t` here?